### DISCUSSION TITLE
power of two much slower than multiplication with itself

### DISCUSSION THREAD

**Discussion by brandy125** [gnudatalanguage/gdl#1350] [Ideas]
Why the power of two (or even one) is so slow in GDL compared to the simple multipliction of two numbers?

```
GDL>s=dindgen(80000)
GDL>tt=systime(1) & for a=1L, 1e4 do begin d=s*s & endfor& print,systime(1)-tt
      0.26803207
GDL> tt=systime(1) & for a=1L, 1e4 do begin d=s^2 & endfor& print,systime(1)-tt
       12.031024
GDL> tt=systime(1) & for a=1L, 1e4 do begin d=s^1 & endfor& print,systime(1)-tt
       12.018444
GDL> tt=systime(1) & for a=1L, 1e4 do begin d=s & endfor& print,systime(1)-tt
      0.23920417
```

Couln't the power of two or power of one be catched by the interpreter and translated to the fast way?
The power of two is being used a lot in maths.

**Comment by GillesDuvert** [comment]
Amusing, there is a special routine in GDL for integer exponents, but not used.
Your suggestion is good, doing it at the interpreter level may be tricky however (anybody ANTLR specialist around?)
converted to issue #1351
